### ISSUE TITLE
missing Date and Message-Id headers on SMTP email delivery

### ISSUE THREAD

**Issue by goenning** [getfider/fider#569]
**Problem:** Spam Assassin doesn't like emails without Date and Message-Id headers, which is exactly what Fider currently does.

**Solution:** We need to set both headers for every email sent via SMTP. As we don't have any requirement for ingesting e-mails, we don't really care about the Message-Id that we use. So for now we could generate a random string with a reasonable length (128?!).

For reference:
- https://wiki.apache.org/spamassassin/Rules/MISSING_MID
- https://wiki.apache.org/spamassassin/Rules/MISSING_DATE

**Comment by lukechampine** [comment]
I can take a stab at this if you point me towards which file(s) it would touch. :)

**Comment by goenning** [comment]
Nice! You'll only need to change `fider/app/pkg/email/smtp/smtp.go` and update the unit tests (smpt_test.go) accordingly. Thanks! :)

**Comment by goenning** [comment]
@lukechampine have you started working on it? @my3 just sent a PR for it

**Comment by lukechampine** [comment]
ah, too slow! I won't bother sending a duplicate PR, then -- although I would suggest using `crypto/rand` + `hex.EncodeToString` instead of importing a `uuid` package.

**Comment by goenning** [comment]
@lukechampine how is that better than using `math/rand` for a random string generator? I understand that crypto/rand is used for operations that require a cryptographically secure random number. But for our use case, would that provide any benefit? What we need here is a unique string generator (like `uuid`) or a long random string generator. We have this function that I was hoping we could use, instead of importing another external package.

https://github.com/getfider/fider/blob/ae9f3d48506a5e9c8a706a2d630a8102661c15ed/app/pkg/rand/random.go#L12-L26

But if there's a reason crypto/rand+hex would produce better results, i'm all in for using it.

**Comment by lukechampine** [comment]
One issue with `math/rand` is seeding. By default, it will be seeded with 0, so every user will be generating the same numbers. Seeding with the current time will probably alleviate this, but...why risk it?

`hex` is just for convenience, so you can write:
```go
buf := make([]byte, 16)
rand.Read(buf)
return hex.EncodeToString(buf)
```

**Comment by my3** [comment]
I was about to say that `uuid` also uses `crypto`. Thought uuid generation will most likely be needed in other places as well and so included that package. If we decide to go with `crypto`, I can make those changes.

**Comment by my3** [comment]
@lukechampine In case you already have code with `crypto`, you can raise a PR and we can close current one.

**Comment by goenning** [comment]
We seed from `time.Now().UnixNano()`. I'm ok with using crypto, it'd rather avoid `uuid` package for now if we can. The MessageID doesn't need to be universally unique.

**Comment by goenning** [comment]
Fixed on #573
